datastream/dataset.py

```diff
@@ -87,6 +87,17 @@ def from_dataframe(dataframe: pd.DataFrame) -> Dataset[pd.Series]:
             get_item=lambda df, index: df.iloc[index],
         )
 
+    def new_columns(self, **kwargs):
```

Comment:
It seems like we can type **kwargs

`def new_columns(self, **kwargs: Callable[pd.Dataframe, pd.Series]):`